helix-core/src/main/java/org/apache/helix/controller/pipeline/PipelineSwitchException.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.helix.controller.pipeline;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Used to exit the current controller pipeline and switch to another pipeline.
+ */
+public class PipelineSwitchException extends StageException {
```

Comment:
Switch is quite confusing and not apply to all cases (for example, entering Maintenance mode), how about PipelineEarlyTerminateException?

Moreover, I think this should be a temporary solution. Using Exception for regular pipeline branches (or alternative logic switches) is very expensive. It would be better to let the pipeline support gracefully switch by returning a execute result, or even support DAG in pipeline.
Of course, not in this PS. TODO here would be enough.

Reply:
Agreed, we should have some logic to more peacefully switch, instead of throwing an exception. Will add a TODO.

Reply:
How about the naming suggestion that I commented on?

Reply:
Discussed with Jiajunwang. I thought PipelineEarlyTerminateException is kind of duplicate with StageException , because StageException is also thrown to early terminate the pipeline. So I wanted a more specific PipelineSwitchException for our case here, which is smaller set of stageException.
We'll keep the current name. If there is a need to change in the future, we can easily refactor it without breaking backward compatibility.